app/shared/models/git_repo.rb

```diff
@@ -60,6 +61,18 @@ class GitRepo
 
     class << self
       attr_accessor :git_action_queue
+
+      # Loads the octokit cache stack for speed-up calls to github service.
+      # As explained in: https://github.com/octokit/octokit.rb#caching
+      def load_octokit_cache_stack
+        @stack ||= Faraday::RackBuilder.new do |builder|
+          builder.use(Faraday::HttpCache, serializer: Marshal, shared_cache: false)
+          builder.use(Octokit::Response::RaiseError)
+          builder.adapter(Faraday.default_adapter)
+        end
+        return if Octokit.middleware.handlers.include?(Faraday::HttpCache)
```

Comment:
That's how this is done, memoized cache 😄 @taquitos

Reply:
Duh, yup. Ignore me. 🤦‍♂️